presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -2382,6 +2382,26 @@ public void dropMaterializedView(ConnectorSession session, SchemaTableName viewN
         }
     }
 
+    @Override
+    public List<String> getValidRefreshMaterializedViewFilterColumns(ConnectorSession session, SchemaTableName viewName)
```

Comment:
We don't need this method. The presto-main can directly call `getMaterializedView` to get the partition mapping. The mapping will tell you which columns you can filter on.

Reply:
If I'm looking at the right place, ConnectorMaterializedViewDefinition returned by getMaterializedView contains columnMappings but not partition mapping. (ConnectorMaterializedViewDefinition is under spi, and thus would not have the concept of partition)

Reply:
We don't need partition mapping. Column mapping actually is better. As long as we have the WHERE on those columns present in the mapping, we are good.

Reply:
In the case of Hive connector, I think the where clause should include a partition and should only include partitions, but not non-partition columns, for performance and correctness. Ex., suppose the mv partition list is ["nation", "region"], and a non-partition column of mv is "customer".

1) If WHERE customer = 'John', that is w/ no partition, it might potentially refresh the entire mv table.

2) If WHERE nation = 'US' AND region = 'NA' AND customer = 'John', refresh would overwrite the entire partition "nation=US/region=NA" but w/ filter custom = 'John' and thus fewer records, compared to the corresponding partition of base tables. So consistency is not maintained.

I also hesitated to add a new API to Metadata at first. But seems no other way. Different connectors might have different valid filter columns.

Reply:
Frankly, we don't have to. The column mapping is already for partitions only. If they are not, we should fix `ColumnMapping` for hive connector.

Reply:
I would take the suggestion and put the partition info into ConnectorMaterializedViewDefinition, in order to avoid adding this Metadata API. It's just I have to repack it at the level of HiveMetadata after creating it the first time in CreateMaterializedViewTask, since we can only get the partition info at the connector HiveMetadata level.